src/Repository/PostRepository.php

```diff
@@ -110,4 +104,22 @@ private function extractSearchTerms(string $searchQuery): array
             return 2 <= mb_strlen($term);
         });
     }
+
+    private function createPaginator(Query $query, int $currentPage, int $pageSize = Post::NUM_ITEMS)
+    {
+        $paginator = new Paginator($query);
+        $hasPreviousPage = $currentPage > 1;
+        $hasNextPage = ($currentPage * $pageSize) < $paginator->count();
+
+        return [
+            'results' => $paginator->getIterator(),
+            'currentPage' => $currentPage,
+            'hasPreviousPage' => $hasPreviousPage,
+            'hasNextPage' => $hasNextPage,
+            'previousPage' => $hasPreviousPage ? $currentPage - 1 : null,
+            'nextPage' => $hasNextPage ? $currentPage + 1 : null,
+            'numPages' => (int) ceil($paginator->count() / $pageSize),
+            'haveToPaginate' => $paginator->count() > $pageSize,
```

Comment:
We call `$paginator->count()` 3 times here, will it execute 3 queries? Maybe put it on a var:
`$count = $paginator->count();`